### PR TITLE
Update CLI reference

### PR DESCRIPTION
**Pull request by tpmccallum**
Signed-off-by: tpmccallum tim.mccallum@fermyon.com

This is in relation to issue https://github.com/fermyon/developer/issues/262

Content must go through a pre-merge checklist.

## Pre-Merge Content Checklist

This documentation has been checked to ensure that:

- [x] The `title, `template`, and `date` are all set
- [x] File does not use CRLF, but uses plain LF (hint: use `cat -ve <filename> | grep '^M' | wc -l` and expect 0 as a result) 
- [x] Has passed [`bart check`](https://bartholomew.fermyon.dev/quickstart)

Screenshot 2023-01-31 at 4 14 13 pm

- [x] Has run `npm run build-index`
- [x] Has been manually tested by running in Spin/Bartholomew (hint: use `PREVIEW_MODE=1` and run `npm run styles` to update styling)

Screenshot 2023-01-31 at 4 17 32 pm

- [x] Headings are using Title Case
- [x] Code blocks have the programming language set to properly highlight syntax and the proper copy directive
- [x] Have tested with `npm run test` and resolved all errors

### CODE REVIEW

**Review by itowlson — CHANGES_REQUESTED** [review]
We should rework the `up` section to distinguish trigger-specific flags from common ones.  At the moment we list the HTTP case, which I'm sure is most common, but users also run it with Redis apps or with no application at all.

File: content/cloud/cli-reference.md [hunk 1]
```diff
@@ -510,4 +593,49 @@ OPTIONS:
...
+            IP address and port to listen on
+            
+            [default: 127.0.0.1:3000]
+
+        --runtime-config-file <RUNTIME_CONFIG_FILE>
+            Configuration file for config providers and wasmtime config
+            
+            [env: RUNTIME_CONFIG_FILE=]
+
+        --tls-cert <TLS_CERT>
```
[the remark this answers is not in the document]
Comment (tpmccallum) [reply content/cloud/cli-reference.md:644]: Thanks @itowlson 
This has been updated now.

File: content/cloud/cli-reference.md [hunk 2]
```diff
@@ -510,4 +593,49 @@ OPTIONS:
...
+        --follow <FOLLOW_ID>
+            Print output for given component(s) to stdout/stderr
+
+        --follow-all
+            Print all component output to stdout/stderr
+
+    -L, --log-dir <APP_LOG_DIR>
+            Log directory for the stdout and stderr of components
+
+        --listen <ADDRESS>
```
[the remark this answers is not in the document]
Comment (tpmccallum) [reply content/cloud/cli-reference.md:634]: Thanks @itowlson 
This has been updated now.

File: content/cloud/cli-reference.md [hunk 3]
```diff
@@ -475,11 +557,12 @@ OPTIONS:
 
 ### Up
 
+Applications without triggers will accept any of the following options. The trigger-specific options are over and above what is shown directly below. Please see the next section for additional trigger options.
```
Comment (itowlson) [inline content/cloud/cli-reference.md:560]: Sorry, I explained this badly.  There are no applications without triggers, and `spin up` won't accept _any_ options without an application.  (But we have to do _something_ when a user types `spin up --help` without an application, which is where the confusion arises.)  It's more like "The following options are always available", _plus_ any trigger-specific options if an application is present.

**Review by itowlson — APPROVED** [review]

**Pull request merged**